src/lib.rs

```diff
@@ -317,67 +317,32 @@ impl<A: Send + 'static> Cancel<A> for Option<A> {
     }
 }
 
-/*
- *
- * ===== Async implementations =====
- *
- */
-
-macro_rules! async_impl_body {
-    ($ty:ty) => (
-        fn is_ready(&self) -> bool {
-            true
-        }
+/// Convenience implementation for (), to ease use of side-effecting functions returning unit
```

Comment:
I took the liberty of adding a comment here so you don't have to explain it to the next person.

I'm not 100% certain whether Rust refers to `()` as void or unit though, so shout out if void is the correct term instead.